src/libos/src/signal/do_sigsuspend.rs

```diff
@@ -0,0 +1,46 @@
+use super::constants::*;
+use super::do_sigprocmask::do_rt_sigprocmask;
+use super::do_sigtimedwait::PendingSigWaiter;
+use super::{sigset_t, MaskOp, SigNum, SigSet, Signal};
+use crate::prelude::*;
+
+pub fn do_sigsuspend(mask: &SigSet) -> Result<()> {
+    debug!("do_sigsuspend: mask: {:?}", mask);
+
+    let thread = current!();
+    let process = thread.process().clone();
+
+    // Set signal mask
+    let mut curr_mask = thread.sig_mask().write().unwrap();
+    let prev_mask = *curr_mask;
+    let update_mask = {
+        let mut set = *mask;
+        // According to man pages, "it is not possible to block SIGKILL or SIGSTOP.
+        // Attempts to do so are silently ignored."
+        set -= SIGKILL;
+        set -= SIGSTOP;
+        set
+    };
+    *curr_mask = update_mask;
+    drop(curr_mask);
+
+    // Suspend for interest signal
+    let interest = SigSet::from_c(!update_mask.to_c());
+    let pending_sig_waiter = PendingSigWaiter::new(thread.clone(), process, interest);
+
+    let err = match pending_sig_waiter.suspend() {
+        Ok(_) => {
+            errno!(EINTR, "Wait for EINTR signal successfully")
```

Comment:
why is `errno!`

Reply:
`sigsuspend() always returns -1, with [errno](https://man7.org/linux/man-pages/man3/errno.3.html) set to indicate the
       error (normally, EINTR).`
https://man7.org/linux/man-pages/man2/sigsuspend.2.html